jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskcomments/TaskCommentsPresenter.java

```diff
@@ -87,81 +78,79 @@ public IsWidget getView() {
+                    @Override
+                    public void callback(Long response) {
+                        refreshComments();
+                        view.clearCommentInput();
+                    }
+                },
+                new ShowErrorInModalCallback()
+        ).addComment(currentTaskId, text, identity.getIdentifier(), addedOn);
     }
 
-    public void removeTaskComment( long commentId ) {
-        taskCommentsServices.call( new RemoteCallback<Long>() {
-            @Override
-            public void callback( Long response ) {
-                refreshComments();
-                view.getNewTaskCommentTextArea().setText( "" );
-                view.displayNotification( "Comment Deleted!" );
-            }
-        }, new ErrorCallback<Message>() {
-            @Override
-            public boolean error( Message message,
-                                  Throwable throwable ) {
-                ErrorPopup.showMessage( "Unexpected error encountered : " + throwable.getMessage() );
-                return true;
-            }
-        } ).deleteComment( currentTaskId, commentId );
+    public void removeTaskComment(long commentId) {
+        taskCommentsServices.call(
+                new RemoteCallback<Long>() {
+                    @Override
+                    public void callback(Long response) {
+                        refreshComments();
+                        view.clearCommentInput();
+                        view.displayNotification("Comment Deleted!");
+                    }
+                },
+                new ShowErrorInModalCallback()
+        ).deleteComment(currentTaskId, commentId);
     }
 
-    public void addDataDisplay( final HasData<CommentSummary> display ) {
-        dataProvider.addDataDisplay( display );
+    public void addDataDisplay(final HasData<CommentSummary> display) {
+        dataProvider.addDataDisplay(display);
     }
 
-    public void onTaskSelectionEvent( @Observes final TaskSelectionEvent event ) {
-        this.currentTaskId = event.getTaskId();
+    public void onTaskSelectionEvent(@Observes final TaskSelectionEvent event) {
+        currentTaskId = event.getTaskId();
         refreshComments();
-        view.getDataGrid().redraw();
+        view.redrawDataGrid();
     }
 
-    public void onTaskRefreshedEvent( @Observes final TaskRefreshedEvent event ) {
-        if ( currentTaskId == event.getTaskId() ) {
+    public void onTaskRefreshedEvent(@Observes final TaskRefreshedEvent event) {
+        if (currentTaskId == event.getTaskId()) {
             refreshComments();
-            view.getDataGrid().redraw();
+            view.redrawDataGrid();
+        }
+    }
+
+    private static class ShowErrorInModalCallback implements ErrorCallback<Message> {
+
+        @Override
+        public boolean error(Message message, Throwable throwable) {
+            ErrorPopup.showMessage("Unexpected error encountered : " + throwable.getMessage());
+            return true;
```

Comment:
+1 to @jhrcek comment below, we need a generic class to use in all the error callbacks. Probably there is already one in Uberfire Extension that we can use.